1xx-enable-ssl-for-kafka-bridge.md

```diff
@@ -0,0 +1,148 @@
+
+Once SSL is enabled, HTTP Bridge server will no longer accept unencrypted connections from external clients. If user wants to allow both HTTP and HTTPS connections, the recommendation would be to deploy 2 separate HTTP Bridge instances, one is enabled with SSL and the other one is not. That way external clients can make both encrypted and unencrypted connections for Kafka operations.
+
+### HTTP Bridge managed by Strimzi operator
+
+When deploying KafkaBridge custom resources with the Strimzi operator, users can set these configurations via `spec.http` in their KafkaBridge CR. Users can enable SSL by setting the new field `spec.http.sslEnable` which is set to false by default. When it is set to true, users have to provide the certificate and key in PEM format via Kubernetes secret that is referenced in the `certificateAndKey` field.
+
+Let's take a look at an example of KafkaBridge CR with the new fields to enable SSL:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 8443
+    tls: #NEW FIELD
+      serverCertChainAndKey: #NEW FIELD
+        secretName: my-secret
+        certificate: public.crt
+        key: private.key
+...
+```
+The liveness and readiness probes will connect to port 8081 (instead `http.port` or 8080). If users have metrics set up, they would need to use port 8081 to connect to the `metrics` endpoint and cannot reconfigure this port number as it's only for internal use.
+
+An example of KafkaBridge CR with other possible SSL configurations:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 8443
+    tls: #NEW FIELD
+      serverCertChainAndKey: #NEW FIELD
+        secretName: my-secret
+        certificate: public.crt
+        key: private.key
+      config: #NEW FIELD
+        ssl.enabled.protocols: "TLSv1.1, "TLSv1.2", "TLSv1.3"
+        ssl.enabled.cipher.suites: "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"
+...
+```
+
+## Future Improvements
+
+### Support more formats for the keystore
+We can support formats such as `PKSC12` and `JKS` as they are common for Java applications. If we support these formats, then we would need to add the following configurations:
+- http.ssl.keystore.type
+- http.ssl.keystore.password
+
+### Support mTLS
+In the future, we can support mutual TLS for the client connections to HTTP Bridge as well. This could be implemented by adding a field `trustedCertificates`, for example:
```

Comment:
```suggestion
In the future, we can support mutual TLS for the client connections to HTTP Bridge as well. This could be implemented by adding a new `trustedCertificates` property in the `tls` section, for example:
```